File: src/com/funivan/phpstorm/refactoring/FindMagicMethods/Visitors/BaseElementVisitor.java
package com.funivan.phpstorm.refactoring.FindMagicMethods.Visitors;

import com.funivan.phpstorm.refactoring.FindMagicMethods.ElementResultCollector;
import com.funivan.phpstorm.refactoring.FindMagicMethods.MagicElementVisitor;
import com.intellij.psi.PsiRecursiveElementWalkingVisitor;

/**
 * @author Ivan Scherbak <dev@funivan.com>
 */
abstract public class BaseElementVisitor extends PsiRecursiveElementWalkingVisitor implements MagicElementVisitor {
    private ElementResultCollector filter;

    @Override
    public void setResultCollector(ElementResultCollector filter) {
        this.filter = filter;
    }

    public ElementResultCollector getResultCollector() {
        return filter;
    }
}

File: src/com/funivan/phpstorm/refactoring/findMagicMethods/Visitors/BaseElementVisitor.kt
package com.funivan.phpstorm.refactoring.findMagicMethods.Visitors

import com.funivan.phpstorm.refactoring.findMagicMethods.MagicElementVisitor
import com.funivan.phpstorm.refactoring.findMagicMethods.results.ResultCollectorInterface
import com.funivan.phpstorm.refactoring.findMagicMethods.results.NullableResultCollector
import com.intellij.psi.PsiRecursiveElementWalkingVisitor

/**
 * @author Ivan Shcherbak alotofall@gmail.com>
 */
abstract class BaseElementVisitor : PsiRecursiveElementWalkingVisitor(), MagicElementVisitor {
    override var collector: ResultCollectorInterface = NullableResultCollector()
}
